modules/ApprovePage.js

```diff
@@ -0,0 +1,26 @@
+const restClient = new mw.Rest();
+
+function sendApprovalRequest( pageId, approve ) {
+	const endpoint = '/page-approvals/v0/page/' + pageId + '/' + (approve ? 'approve' : 'unapprove');
+	const csrfToken = mw.user.tokens.values.csrfToken;
+
+	restClient.post( endpoint, {
+		token: csrfToken
+	} ).then( function( response ) {
+		console.log( 'Request successful:', response );
+		mw.notify( 'Request successful: ' + JSON.stringify( response ) );
+	} ).catch( function( error ) {
+		console.error( 'API request failed:', error );
+		mw.notify( 'API request failed: ' + error, { type: 'error' } );
+	} );
+}
+
+$( document ).on( 'click', '#approveButton', function() {
+	const pageId = $( this ).data( 'page-id' );
```

Comment:
Page ID is already available in the page:

`mw.config.get( 'wgArticleId' );`

See https://www.mediawiki.org/wiki/Manual:Interface/JavaScript#All_pages_(user/page-specific)